source/common/stats/thread_local_store.cc

```diff
@@ -137,6 +137,7 @@ void ThreadLocalStoreImpl::initializeThreading(Event::Dispatcher& main_thread_di
   tls_->set([](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
     return std::make_shared<TlsCache>();
   });
+  tls_->runOnAllThreads([this, &tls] { tls.dispatcher().initializeStats(*this); });
```

Comment:
Note that it would, in theory, be a bit cleaner to be able to do this in the callback passed to `set` above, since that callback is given a dispatcher, but it doesn't work due to some tricky ordering issues.

Reply:
I'd be interested in hearing this story :) What you have seems pretty clean though, but needs comments for posterity.